cluster-autoscaler/utils/taints/taints.go

```diff
@@ -416,3 +442,50 @@ func FilterOutNodesWithStartupTaints(taintConfig TaintConfig, allNodes, readyNod
 	}
 	return newAllNodes, newReadyNodes
 }
+
+// CountNodeTaints counts used node taints.
+func CountNodeTaints(nodes []*apiv1.Node, taintConfig TaintConfig) map[string]int {
+	foundTaintsCount := make(map[string]int)
+	for _, node := range nodes {
+		for _, taint := range node.Spec.Taints {
+			key := getTaintTypeToReport(taint.Key, taintConfig)
+			if _, ok := foundTaintsCount[key]; ok {
+				foundTaintsCount[key] += 1
+			} else {
+				foundTaintsCount[key] = 1
+			}
+		}
+	}
+	return foundTaintsCount
+}
+
+func getTaintTypeToReport(key string, taintConfig TaintConfig) string {
+	// Track deprecated taints.
+	if strings.HasPrefix(key, IgnoreTaintPrefix) {
+		return IgnoreTaintPrefix
+	}
+
+	if _, ok := explicitlyReportedNodeTaints[key]; ok {
+		return key
+	}
+
+	if _, ok := taintConfig.StartupTaints[key]; ok {
```

Comment:
optional: There's already 3-4 places in this file that separately check prefix and exact values in `taintConfig`. WDYT about slightly refactoring this to make fields in `TaintConfig` private and expose `IsStartupTaint`/`IsStatusTaint` functions instead?

Reply:
Added